src/engine/enginemaster.cpp

```diff
@@ -366,16 +406,80 @@ void EngineMaster::process(const int iBufferSize) {
     // Mix all the PFL enabled channels together.
     m_headphoneGain.setGain(chead_gain);
 
-    if (m_bRampingGain) {
-        ChannelMixer::mixChannelsRamping(
-                m_headphoneGain, &m_activeHeadphoneChannels,
-                &m_channelHeadphoneGainCache,
-                m_pHead, iBufferSize);
+    // Mixxx gets input signals delayed by the input latency,
+    // so to preserve the relative timing of the input with the master and headphones
+    // outputs, delay the master, booth, and headphones outputs by the input latency.
+    // This way, if the microphone user is on beat with the output they hear coming from
+    // the speakers/headphones, the input will remain on beat at the expense of
+    // adding some latency.
```

Comment:
A delay that is is only applied to the record/broadcast signal is necessary for direct monitoring, otherwise the record/broadcast signal is off beat when the signal to the speakers is on beat.